chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/PendingSurveillanceManagerImpl.java

```diff
@@ -0,0 +1,715 @@
+        try {
+            surveillanceThresholdToProcessAsJob = Integer.parseInt(surveillanceThresholdToProcessAsJobStr);
+        } catch (final NumberFormatException ex) {
+            LOGGER.error(
+                    "Could not format " + surveillanceThresholdToProcessAsJobStr + " as an integer. Defaulting to"
+                            + " 50 instead.");
+        }
+        return surveillanceThresholdToProcessAsJob;
+    }
+
+    private void validate(final Surveillance surveillance) {
+        validator.validate(surveillance, false);
+    }
+
+    private void checkSurveillanceAuthority(final Surveillance surv) throws SurveillanceAuthorityAccessDeniedException {
+        Boolean hasOncAdmin = Util.isUserRoleAdmin() || Util.isUserRoleOnc();
+        Boolean hasAcbAdmin = Util.isUserRoleAcbAdmin();
+        if (StringUtils.isEmpty(surv.getAuthority())) {
+            // If user has ROLE_ADMIN and ROLE_ACB
+            // return 403
+            if (hasOncAdmin && hasAcbAdmin) {
+                String errorMsg = "Surveillance cannot be created by user having " + Authority.ROLE_ADMIN + " and "
+                        + Authority.ROLE_ACB;
+                LOGGER.error(errorMsg);
+                throw new SurveillanceAuthorityAccessDeniedException(errorMsg);
+            }
+        } else {
+            // Cannot have surveillance authority as ROLE_ADMIN for user lacking
+            // ROLE_ADMIN
+            if (surv.getAuthority().equalsIgnoreCase(Authority.ROLE_ADMIN) && !hasOncAdmin) {
+                String errorMsg = "User must have authority " + Authority.ROLE_ADMIN;
+                LOGGER.error(errorMsg);
+                throw new SurveillanceAuthorityAccessDeniedException(errorMsg);
+            } else if (surv.getAuthority().equalsIgnoreCase(Authority.ROLE_ACB)) {
+                // Cannot have surveillance authority as ACB for user lacking ONC
+                // and ACB roles
+                if (!hasOncAdmin && !hasAcbAdmin) {
+                    String errorMsg = "User must have ONC or ACB roles for a surveillance authority created by ACB";
+                    LOGGER.error(errorMsg);
+                    throw new SurveillanceAuthorityAccessDeniedException(errorMsg);
+                }
+            }
+        }
+    }
+
+    private Long createSurveillance(final Surveillance surv)
+            throws UserPermissionRetrievalException, SurveillanceAuthorityAccessDeniedException {
+        Long insertedId = null;
+        checkSurveillanceAuthority(surv);
+        updateNullAuthority(surv);
+
+        try {
+            insertedId = survDao.insertSurveillance(surv);
+        } catch (final UserPermissionRetrievalException ex) {
+            LOGGER.error("Error inserting surveillance.", ex);
+            throw ex;
+        }
+
+        return insertedId;
+    }
+
+    private void updateNullAuthority(final Surveillance surv) {
+        Boolean hasOncAdmin = Util.isUserRoleAdmin();
```

Comment:
What happens to a user who is ROLE_ONC?

Reply:
This method is not needed.  Whatever is set for the authority is overridden in the DAO.  This method has been removed.